### ISSUE TITLE
#section recursion

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#550]
the page en.wikipedia.org/wiki/National_Popular_Vote_Interstate_Compact gives two errors

    invoke failed: National_Popular_Vote_Interstate_Compact {{#invoke:NPVIC status|overlays|section = passed ...
    invoke failed: National_Popular_Vote_Interstate_Compact {{#invoke:NPVIC status|overlays|section = pending ...
However, I believe the error is related to \#section 

The wikitext of the pages contains the lines

    <section begin=As_of_month/>{{as of|2019|5}}<section end=As_of_month/>
    <section begin=passed/>MD, NJ, IL, HI, WA, MA, DC, VT, CA, RI, NY, CT, CO, DE, NM<section end=passed/>
    <section begin=pending/>AZ, GA, KS, MN, NC, NV, NH, OH, OR, SC, WI<section end=pending/>
and further down is the wikitext

    {{#section:{{PAGENAME}}|As_of_month}}}}

The way that \#section is processed is that the page is fetched from the database (or cache),
then it is expanded,
all the \<section entries are identified and this in turn is cached

Unfortunately for this page that process does not work well
In this case, the page is fetched, expanded and during the expansion it tries to do another \#section extract (so a loop)

This comes about because the page is self referencing

There is some loop logic that kicks in to stop infinite recursion, but this is not the correct way to do it (hence the errors)

If it could be guaranteed that \<section is not generated by template, then a solution could be to identify the \<section entries (as done now) and at the point of identifying the section range perform the expansion

**Comment by gnosygnu** [comment]
Yeah, I think this is going to require a fundamental change to fix. This falls in the same category of "parser template" bugs which I've mentioned on other issues. Ultimately, I think this will only be fixed when I do the mediawiki rewrite in #508 

I created a ```[backlog - parser - template]``` tag and will add them retroactively to others.

Thanks.
